Mathlib/NumberTheory/Transcendental/AnalyticPart.lean

```diff
@@ -0,0 +1,257 @@
+      have hf'' : derivWithin (fun _ ↦ 0) (Metric.ball z₀ r) x =
+        derivWithin f (Metric.ball z₀ r) x := by
+        apply Filter.EventuallyEq.derivWithin_eq_of_nhds
+        rw [Filter.EventuallyEq, Filter.eventually_iff_exists_mem]
+        use Metric.ball z₀ r
+        refine ⟨IsOpen.mem_nhds Metric.isOpen_ball hx, fun z a ↦ ?_⟩
+        · exact (Complex.ext (congrArg Complex.re (hf' a)) (congrArg Complex.im (hf' a))).symm
+      rw [← derivWithin_of_mem_nhds (h := IsOpen.mem_nhds Metric.isOpen_ball hx)]
+      aesop
+
+lemma analyticOrderAt_eq_succ_iff_deriv_order_eq_pred (f : ℂ → ℂ) z₀ (hf : AnalyticAt ℂ f z₀)
+  (n : ℕ) : f z₀ = 0 → analyticOrderAt (deriv f) z₀ = (n - 1 : ℕ) →
+      n > 0 → analyticOrderAt f z₀ = n := by
+    intros hzero horder hn
+    have : ∃ m, analyticOrderAt f z₀ = m := by simp
+    obtain ⟨m, Hn'⟩ := this
+    cases m
+    · exfalso
+      have ht := (analyticOrderAt_deriv_eq_top_iff_of_eq_zero z₀ f hf hzero).2 Hn'
+      exact (ENat.coe_ne_top (n - 1)) (by grind)
+    · rename_i n'
+      cases n'
+      · exfalso; exact ((AnalyticAt.analyticOrderAt_eq_zero hf).1 Hn') hzero
+      · rename_i n''
+        have hnn := Complex.analyticOrderAt_deriv_of_pos hf (n := n'' + 1) Hn' (by grind)
+        simp only [horder] at hnn
+        have : n = n'' + 1 := by norm_cast at hnn; grind
+        rw [this]
+        exact Hn'
+
+--Might be necessary (hf : ∀ z : ℂ, AnalyticAt ℂ R z)
+lemma iterated_deriv_mul_pow_sub_of_analytic (r : ℕ) {z₀ : ℂ} {R R₁ : ℂ → ℂ}
+   (hf1 : ∀ z : ℂ, AnalyticAt ℂ R₁ z) (hR₁ : ∀ z, R z = (z - z₀)^r * R₁ z) :
+  ∀ k ≤ r ,
+    ∃ R₂ : ℂ → ℂ, (∀ z : ℂ, AnalyticAt ℂ R₂ z) ∧ ∀ z, deriv^[k] R z =
+   (z - z₀)^(r-k) * (r.factorial/(r-k).factorial * R₁ z + (z-z₀)* R₂ z) := by
+      intros k hkr
+      induction k
+      · use 0
+        simp only [Function.iterate_zero, id_eq, tsub_zero, Pi.zero_apply, mul_zero, add_zero]
+        refine ⟨fun z ↦ ?_, fun z ↦ ?_⟩
+        · refine Differentiable.analyticAt (differentiable_zero) z
+        · rw [hR₁ z]
+          simp only [mul_eq_mul_left_iff, pow_eq_zero_iff', ne_eq]
+          left
+          rw [div_self]
+          · grind
+          · simp only [ne_eq, Nat.cast_eq_zero]
+            exact Nat.factorial_ne_zero r
+      · rename_i k IH
+        simp only [Function.iterate_succ, Function.comp_apply]
+        have change_deriv (R : ℂ → ℂ) (z : ℂ) :
+          deriv^[k] (deriv R) z = deriv (deriv^[k] R) z := by
+          have : deriv^[k] (deriv R) z = deriv^[k+1] R z := by aesop
+          have : deriv (deriv^[k] R) z = deriv^[k+1] R z := by
+            induction k
+            · aesop
+            · rename_i k
+              simp only [Function.iterate_succ, Function.comp_apply]
+              simp only [Function.iterate_succ, Function.comp_apply] at IH
+              rw [← iteratedDeriv_eq_iterate] at *
+              rw [← iteratedDeriv_succ, this]
+              simp only [Function.iterate_succ, Function.comp_apply]
+          rw [this, ← this]
```

Comment:
I broke down when I got to this line, I feel like this is a bit disrespectful of reviewers' time.
Please go through your own code and clean it up to the best of your ability before submitting it for a review. You can mark a PR as a work-in-progress by commenting "WIP".